File: 53-MaximumSubarray/53-MaximumSubarray.py
# Last updated: 8/9/2026, 12:37:40 PM
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        s=maxi=nums[0]
        for i in nums[1:]:
            s=max(s,0)+i
            maxi=max(maxi,s)
        return maxi

File: 53-MaximumSubarray/53-MaximumSubarray.java
// Last updated: 8/9/2026, 12:37:45 PM
class Solution {
    public int maxSubArray(int[] nums) {
        int sum=Integer.MIN_VALUE;
        int s=0;
        for (int i=0;i< nums.length;i++){
            
                s=s+nums[i];
                sum=Math.max(sum,s);      
                if(s<0){
                    s=0;
                }         
            
        }
        return sum;
    }
}